### DISCUSSION TITLE
Desktop files support in `${HOMEBREW_PREFIX}/share/applications`

### DISCUSSION THREAD

**Discussion by tranzystorekk** [Homebrew/discussions#3945] [Linux]
Are desktop files installed by formulae somehow supported/documented in homebrew? I wanted to know if there's a trick to e.g. getting desktop shortcuts in any desktop environment.

**Comment by SMillerDev** [comment]
They are not afaik.

**Comment by tom-spalding** [comment]
Still relevant today. I think we can just manually create a `.desktop` file in `Formula/kittycatapp.rb`?

Something like this I'm hoping..

`Formula/kittycatapp.rb`:

```ruby
class KittyCatApp < Formula
  # ...

  def install
    # ...

    install_desktop_entry
  end

  # And maybe also this depending on your Linux distro?
  def post_install
    system "update-desktop-database", "#{HOMEBREW_PREFIX}/share/applications" if which("update-desktop-database")
  end

  private

  def install_desktop_entry
    desktop_file = Dir[libexec/"*.desktop"].first
    return unless desktop_file

    entry = File.read(desktop_file)
    entry.gsub!(/^Exec=.*/, "Exec=#{bin}/kittycatapp")
    (share/"applications").mkpath
    (share/"applications/kittycatapp.desktop").write entry

    icon_name = entry[/^Icon=(.+)$/, 1]
    return unless icon_name

    Dir[libexec/"**/#{icon_name}.{png,svg}"].each do |icon_file|
      size = icon_file[%r{/(\d+x\d+)/}, 1] || "512x512"
      icon_dest = share/"icons/hicolor/#{size}/apps"
      icon_dest.mkpath
      icon_dest.install icon_file
    end
  end
end
```

**Reply by SMillerDev** [reply]
For Homebrew core we don't ship f configuration that upstream doesn't ship, but in a tap this seems like a decent option.

**Comment by xPMo** [comment]
I was looking into this today, and when trying to add `.linuxbrew/share` to XDG_DATA_DIRS (which _should_ pull in any .desktop files into a DE) it borked Plasma. Found [this issue](https://github.com/Homebrew/brew/issues/18643), seems like this is unfortunately in limbo atm.
